scipy/stats/tests/test_fit.py

```diff
@@ -335,9 +339,12 @@ def test_basic_fit(self, dist_name):
             ref = ref[:-1]
             ref[-1] = np.floor(loc)
             data = dist.rvs(*ref, size=N, random_state=rng)
-            res = stats.fit(dist, data, bounds[:-1], optimizer=self.opt)
+            bounds = bounds[:-1]
         if getattr(dist, 'pdf', False):
             data = dist.rvs(*ref, size=N, random_state=rng)
+
+        with npt.suppress_warnings() as sup:
+            sup.filter(RuntimeWarning, "overflow encountered")
             res = stats.fit(dist, data, bounds, optimizer=self.opt)
 
         assert_allclose(res.params, ref, **self.tols)
```

Comment:
This resolves a lot of the false positives.
```suggestion
        cond1 = np.allclose(res.params, ref, **self.tols)
        nllf1 = dist.nnlf(res.params, data)
        nllf2 = dist.nnlf(ref, data)
        cond2 = nllf1 < nllf2
        cond3 = np.allclose(nllf1, nllf2, rtol=1e-2)
        assert cond1 or cond2 or cond3
```